### PR TITLE
Remove redundant `checked_cast`, add `utimes` family

### PR DESCRIPTION
**Pull request by carbotaniuman**
As the comment in one of the files says, I'm not sure what Rustix's story is wrt 64bit time on 32-bit systems.

This also adds `some_or_ret_einval`, which I'm down to bikeshed, but it should get rid of all the unwraps for parsing flags from bits.

### CODE REVIEW

**Comment by carbotaniuman** [comment]
Hmmm I thought x86 kernels didn't have 64 bit time support until 5.1, and we were still on 2.x something... Weird. It looks like we're 64-bit time always?

**Comment by sunfishcode** [comment]
Rustix uses 64-bit time types always, and uses the 64-bit APIs when the platform has them. It falls back to the 32-bit APIs when it needs to, though it converts everything to/from 64-bit types, and fails if the conversions ever fail. The idea is to be y2038-compatible when running on kernels that support the APIs to be y2038-compatible.

**Comment by carbotaniuman** [comment]
I am so confused as to how this breaks... It appears to be an ICE in some macro expansion code?

**Comment by carbotaniuman** [comment]
Ah, new Rust nightly ICE... Any reason with Rustix provides both `futimes` and `utimensat`? Just for ergonomic purposes, or is there some kernel difference?

**Comment by sunfishcode** [comment]
Passing a NULL pathname in `utimensat` is not portable outside of Linux, so having a separate `futimens` more closely reflects how the libc backend works. And ergonomics :-).

**Comment by sunfishcode** [comment]
It looks like the ICE might be due to https://github.com/rust-lang/rust/issues/99261.

**Comment by sunfishcode** [comment]
`main` now has pinned a version of nightly; so if you rebase this on main it should be fixed.

File: c-scape/src/time/mod.rs [hunk 1]
```diff
@@ -3,25 +3,81 @@ use core::convert::TryInto;
 use errno::{set_errno, Errno};
 use libc::c_int;
 
+// So 32-bit time support was only added to 64bit in 5.1,
+// so I'm not even sure what Rustix's deal with this is.
+
+// TODO: clarify
```
Comment (sunfishcode) [inline c-scape/src/time/mod.rs:9]: It looks like you've fixed the main problems here. Can you say what the remaining open questions are?

File: c-scape/src/fs/utime.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+unsafe extern "C" fn utimensat(
+    fd: c_int,
+    path: *const c_char,
+    times: *const libc::timespec,
+    flag: c_int,
+) -> c_int {
+    libc!(libc::utimensat(fd, path, times, flag));
+
+    let times = times as *const [libc::timespec; 2];
+    let flags = some_or_ret_einval!(AtFlags::from_bits(flag as _));
```
[the remark this answers is not in the document]
Comment (carbotaniuman) [reply c-scape/src/fs/utime.rs:59]: I feel like we've had this conversation once before, and I think I just forgot about it, my bad. It's just so tempting... I'll switch it back in a bit.

**Comment by sunfishcode** [comment]
Thanks!

**Pull request merged**